packages/avocode-email-tagsinput/test/e2e/avocode-email-tagsinput.spec.js

```diff
@@ -0,0 +1,154 @@
+      expect(editor).to.exist
+    })
+
+    describe('user typing', () => {
+      it('should insert text', async () => {
+        await page.click(tagsInputFieldSelector)
+        await page.type(tagsInputFieldSelector, 'abcd')
+        const text = await getInputText(page)
+        expect(text)
+          .to
+          .equal('abcd')
+      })
+
+      it('should delete text with backspace', async () => {
+        await page.click(tagsInputFieldSelector)
+        await page.type(tagsInputFieldSelector, 'abcd')
+        await page.keyboard.press('Backspace')
+        const text = await getInputText(page)
+        expect(text)
+          .to
+          .equal('abc')
+      })
+
+      it('should delete text with delete', async () => {
+        await page.click(tagsInputFieldSelector)
+        await page.type(tagsInputFieldSelector, 'abcd')
+        await page.keyboard.press('ArrowLeft')
+        await page.keyboard.press('Delete')
+        const text = await getInputText(page)
+        expect(text)
+          .to
+          .equal('abc')
+      })
+
+      it('should navigate one char left', async () => {
+        await page.click(tagsInputFieldSelector)
+        await page.type(tagsInputFieldSelector, 'abcd')
+        await page.keyboard.press('ArrowLeft')
+        const [ anchorOffset ] = await getAnchorAndFocus(page)
+        expect(anchorOffset)
+          .to
+          .equal(3)
+      })
+    })
+
+    it('should not be allowed to add invalid email as a tag', async () => {
+      await addEmailTag(page, 'testemail@ex.11')
+      const tagNodes = await getTagNodes(page)
+      expect(tagNodes)
+        .to
+        .have
+        .length(0)
+    })
+
+    it('should be allowed to add valid email as a tag', async () => {
+      await addEmailTag(page, 'testEmail@example.com')
+      const tagNodes = await getTagNodes(page)
+      expect(tagNodes)
+        .to
+        .have
+        .length(1)
+    })
+
+    describe('user can add new tag typing valid email and hitting comma, enter or space', () => {
+      const validEmailTag = 'test@ex.com'
+
+      it('should be possible to add an email tag hitting Comma key', async () => {
+        await addEmailTag(page, validEmailTag, 'Comma')
+        const tagNodes = await getTagNodes(page)
+        expect(tagNodes)
+          .to
+          .have
+          .length(1)
+      })
+
+      it('should be possible to add an email tag hitting Enter key', async () => {
+        await addEmailTag(page, validEmailTag, 'Enter')
+        const tagNodes = await getTagNodes(page)
+        expect(tagNodes)
+          .to
+          .have
+          .length(1)
+      })
+
+      it('should be possible to add an email tag hitting Space key', async () => {
+        await addEmailTag(page, validEmailTag, 'Space')
+        const tagNodes = await getTagNodes(page)
+        expect(tagNodes)
+          .to
+          .have
+          .length(1)
+      })
+    })
+  })
+  describe('Unique Email Tags Input', () => {
```

Comment:
Please add whitespace here as well.

Reply:
And pleas change description to `unique emails`

Reply:
I will fix those white spaces. I wonder how those spaces appeared! Maybe I accidentally gave the command in my IDE. Sorry about that.

Reply:
Maybe you use Prettier or some auto-fix.

Reply:
Please add new line above `describe`